worlds/civ_6/__init__.py

```diff
@@ -0,0 +1,192 @@
+        super().__init__(multiworld, player)
+        self.location_by_era = generate_era_location_table()
+
+        self.location_table: Dict[str, CivVILocationData] = {}
+        self.item_table = generate_item_table()
+
+        for _era, locations in self.location_by_era.items():
+            for _item_name, location in locations.items():
+                self.location_table[location.name] = location
+
+    def get_filler_item_name(self) -> str:
+        return get_random_filler_by_rarity(self, FillerItemRarity.COMMON, self.item_table).name
+
+    def create_regions(self) -> None:
+        create_regions(self, self.options, self.player)
+
+    def set_rules(self) -> None:
+        if self.options.boostsanity:
+            create_boost_rules(self)
+
+    def create_item(self, name: str) -> Item:
+        item: CivVIItemData = self.item_table[name]
+        classification = item.classification
+        if self.options.boostsanity:
+            if item.civ_name in BOOSTSANITY_PROGRESSION_ITEMS:
+                classification = ItemClassification.progression
+
+        return CivVIItem(item, self.player, classification)
+
+    def create_items(self) -> None:
+        progressive_era_item = None
+        for item_name, data in self.item_table.items():
+            # Don't add progressive items to the itempool here
+            if data.item_type == CivVICheckType.PROGRESSIVE_DISTRICT:
+                continue
+            if data.item_type == CivVICheckType.ERA:
+                # Don't add era items in this way
+                progressive_era_item = data
+                continue
+            if data.item_type == CivVICheckType.GOODY:
+                continue
+
+                # If we're using progressive districts, we need to check if we need to create a different item instead
+            item_to_create = item_name
+            if self.options.progression_style != "none":
+                item: CivVIItemData = self.item_table[item_name]
+                if item.progression_name != None:
+                    item_to_create = self.item_table[item.progression_name].name
+
+            self.multiworld.itempool += [self.create_item(
+                item_to_create)]
+
+        # Era items
+        if self.options.progression_style == "eras_and_districts":
+            # Add one less than the total number of eras (start in ancient, don't need to find it)
+            for era in EraType:
+                if era.value == "ERA_ANCIENT":
+                    continue
+                self.multiworld.itempool += [self.create_item(
+                    progressive_era_item.name)]
+
+        num_filler_items = 0
+        # Goody items, create 10 by default if options are enabled
+        if self.options.shuffle_goody_hut_rewards:
+            num_filler_items += 10
+
+        if self.options.boostsanity:
+            boost_data = get_boosts_data()
+            num_filler_items += len(boost_data)

```

Comment:
```suggestion
        if self.options.boostsanity:
            num_filler_items += len(get_boosts_data())
```
Feel free to not do this one, it's more of personal preference